crates/bioauth-flow-rpc/src/lib.rs

```diff
@@ -366,17 +367,77 @@ where
             )
             .map_err(AuthenticateError::RuntimeApi).map_err(errtype)?;
 
-        self.pool
+        info!("Bioauth flow - submitting authenticate transaction");
+
+        let tx_hash = self.pool.hash_of(&ext);
+
+        let mut watch = self.pool
             .submit_and_watch(
                 &sp_api::BlockId::Hash(at),
                 sp_runtime::transaction_validity::TransactionSource::Local,
                 ext,
             )
             .await
-            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?;
-
-        info!("Bioauth flow - authenticate transaction complete");
-
-        Ok(())
+            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?.fuse();
+
+        tokio::spawn(async move {
+            while let Some(tx_status) = watch.next().await {
+                match tx_status {
```

Comment:
Let's instead do it like this:

```suggestion
            loop {
                let maybe_tx_status = watch.next().await;
                match maybe_tx_status {
```

And match with `Some(TransactionStatus::Finalized((block_hash, _))) ...` - that way we can we can also match `None` in the same way and handle it explicitly as a behaviour case.